lldb/tools/lldb-dap/SourceBreakpoint.h

```diff
@@ -52,14 +37,33 @@ struct SourceBreakpoint : public Breakpoint {
   static bool BreakpointHitCallback(void *baton, lldb::SBProcess &process,
                                     lldb::SBThread &thread,
                                     lldb::SBBreakpointLocation &location);
-};
 
-inline bool operator<(const SourceBreakpoint &lhs,
-                      const SourceBreakpoint &rhs) {
-  if (lhs.line == rhs.line)
-    return lhs.column < rhs.column;
-  return lhs.line < rhs.line;
-}
+  inline bool operator<(const SourceBreakpoint &rhs) {
+    if (line == rhs.line)
+      return column < rhs.column;
+    return line < rhs.line;
+  }
+
+  uint32_t GetLine() const { return line; }
+  uint32_t GetColumn() const { return column; }
+
+protected:
+  // logMessage part can be either a raw text or an expression.
+  struct LogMessagePart {
+    LogMessagePart(llvm::StringRef text, bool is_expr)
+        : text(text), is_expr(is_expr) {}
+    std::string text;
+    bool is_expr;
+  };
```

Comment:
I think other kinds of breakpoints can have log messages. Should we make this into its own helper that is shared between all the BP types?

Reply:
Looking at the spec, seems like this is limited to SourceBreakpoints:

```
  /**
   * The debug adapter supports log points by interpreting the `logMessage`
   * attribute of the `SourceBreakpoint`.
   */
  supportsLogPoints?: boolean;
```

Reply:
Ah, my bad, I'm thinking of `condition` and `hitCondition` that are supported by multiple types of breakpoints.